images/kibana/tests/main.tf

```diff
@@ -0,0 +1,54 @@
+terraform {
+  required_providers {
+    oci       = { source = "chainguard-dev/oci" }
+    imagetest = { source = "chainguard-dev/imagetest" }
+  }
+}
+
+locals {
+  namespace = "kibana"
+  parsed    = provider::oci::parse(var.digest)
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+
+data "imagetest_inventory" "this" {}
+
+
+resource "imagetest_harness_docker" "this" {
+  name      = "kibana"
+  inventory = data.imagetest_inventory.this
+
+  envs = {
+    IMAGE_NAME = var.digest
+    TAG        = local.parsed.pseudo_tag
+  }
+}
+
+resource "imagetest_feature" "this" {
```

Comment:
Is it possible to increase the test coverage here and include some core functionality tests?

Reply:
I tried a few things, the best approximation that it is working is connecting up to an Elasticsearch. Without one available, Kibana just sits waiting to initialize. If Kibana detects an elasticsearch (by default it looks for 127.0.0.1:9200), it autoconfigures itself and spits out the `Kibana is now available` message.

I'll try to figure out a curl command to post something to it, or elasticsearch and validate with another test.

Reply:
Without elasticsearch logs look like this:

```
[2024-05-16T20:19:38.332+00:00][INFO ][http.server.Preboot] http server running at http://localhost:5601
[2024-05-16T20:19:38.375+00:00][INFO ][plugins-system.preboot] Setting up [1] plugins: [interactiveSetup]
[2024-05-16T20:19:38.381+00:00][INFO ][preboot] "interactiveSetup" plugin is holding setup: Validating Elasticsearch connection configuration…
[2024-05-16T20:19:38.392+00:00][INFO ][root] Holding setup until preboot stage is completed.


i Kibana has not been configured.

Go to http://localhost:5601/?code=574708 to get started.
```

Reply:
could we plug in in our ECK operator I wonder?

Reply:
The operator looks unopinionated enough that I think that might just work @mamccorm 🎉  didn't even realize we had it. Will give it a try and see what it does